frontend/src/components/application/Collection.tsx

```diff
@@ -42,11 +45,38 @@ const ApplicationCollection: FunctionComponent<Props> = ({
     ? applications
     : applications.slice((page - 1) * perPage, page * perPage)
 
+  const refresh = onRefresh ? (
+    <Button onClick={onRefresh} className="float-right">
+      {t("Refresh")}
+    </Button>
+  ) : null
+
+  const header = (
+    <span className="flex items-center justify-between">
+      <h2>{title}</h2>
+      {refresh}
+    </span>
+  )
+
+  if (applications.length === 0) {
```

Comment:
This won't work, as we return early above, if we have no application data.

Reply:
Where is the early return? I moved the "No applications to show" message to ApplicationCollection so I wouldn't have to duplicate the refresh button.

Reply:
Line 24 in my the edition I've checked out from your branch

Reply:
I think I confused some files? Can't find the place anymore.